src/_pytest/mark/structures.py

```diff
@@ -266,7 +267,7 @@ def combined_with(self, other: "Mark") -> "Mark":
 # A generic parameter designating an object to which a Mark may
 # be applied -- a test function (callable) or class.
 # Note: a lambda is not allowed, but this can't be represented.
-Markable = TypeVar("Markable", bound=Union[Callable[..., object], type])
+Markable = TypeVar("Markable", bound=Callable[..., Any])
```

Comment:
Any reason to change the `object` to `Any` here?

Reply:
I just double checked an `None` is also an instance of `Any`, but it seems odd to specify `object` when you mean you don't care. I can leave it as is if `object` is preferred and it doesn't trip up type checkers.

Reply:
I'll try to explain:

Both types are compatible with everything, i.e. `x: object = <anything>` and `x: Any = <anything>` both work. The difference is that `x: Any` allows doing anything *with* the object, while `x: object` allows *nothing*. So `object` is completely safe - if you want to do anything with it besides generic object capabilities like `str(x)`, you have to dynamically check its type with `isinstance(x, <type>)`. `Any` however is completely unsafe, and strict typing would conceptually disallow it.

So when to use which? If pytest returns some unknown value to the user, I don't want to force them to do `isinstance` checks, which are an headache, so returning `Any` makes sense. But if pytest *receives* such a value, it should use the safe variant.

In this specific case, the return type is just used in a `TypeVar` which is bound to some concrete value anyway, so whether it's `object` or `Any` shouldn't actually matter. So using the safer `object` is nicer.

Reply:
I'm not sure I agree with which to use based on what I have seen. And based on my overall comment https://github.com/pytest-dev/pytest/pull/8674#issuecomment-841715808 mypy _does_ read into `object` too much.

I see you're basically describing what's here https://mypy.readthedocs.io/en/latest/dynamic_typing.html#any-vs-object, but what you are trying to do is more akin to TypeScripts `unknown` and `any`. Although I understand what you are trying to achieve, I don't think that always works in practice, or at least it touches more rough edges of mypy. I definitely agree with what you're trying to accomplish, but I think since mypy isn't there yet and type checking is still relatively young using `Any` is safer (rough edges not for the users) than using `object` unless there is a PEP which will either add an `unknown` type a la TypeScript.